modules/tide_content_collection/tide_content_collection.module

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * @file
+ * Contains tide_content_collection.module.
+ */
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\paragraphs\ParagraphInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ */
+function tide_content_collection_paragraph_access(ParagraphInterface $entity, $operation, AccountInterface $account) {
+  $type = $entity->getType();
+  if ($type === 'content_collection') {
+    if (in_array($operation, ['update', 'delete'])) {
+      return AccessResult::allowedIfHasPermission($account, 'access content_collection paragraph');
+    }
+  }
+  return AccessResult::neutral();
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_create_access().
+ */
+function tide_content_collection_paragraph_create_access(AccountInterface $account = NULL, array $context = [], $entity_bundle = NULL) {
+  $type = $entity_bundle;
+  if ($type === 'content_collection') {
+    return AccessResult::allowedIfHasPermission($account, 'access content_collection paragraph');
+  }
+  return AccessResult::neutral();
+}
```

Comment:
Hi @joshua-salsadigital 
Just trying to understand the permission scenario here. So looks like it creates a custom permission for the old content_collection paragraph, isn't it ? if the permission is set for specific user role then they can create, update or delete this component. 
So my question is - why only for old content collection and not for the new one and no permission for content_collection_enhanced ?
- Also I have this test branch in [reference site](https://nginx-php.pr-83.content-reference-sdp-vic-gov-au.sdp2.sdp.vic.gov.au/) and the permission is set only administrator and site_admin role but the editor still can add the old component. Unless I am understanding or setting up the permission wrongly. So just wanted to clarify these things.

Reply:
@MdNadimHossain that permission was created after a discussion with @anthony-malkoun. The Raw JSON component (the original component) is too powerful and we don't encourage editors to have access to that component.

I doubt if your reference site has the changes from this PR. The new component doesn't exist on your site: https://nginx-php.pr-83.content-reference-sdp-vic-gov-au.sdp2.sdp.vic.gov.au/admin/structure/types/manage/landing_page/fields/node.landing_page.field_landing_page_component

Reply:
@MdNadimHossain my mistake, it seems that you rename the old component.